openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -78,12 +80,46 @@ def _gather_graded_items(root, due):  # lint-amnesty, pylint: disable=missing-fu
     return []
 
 
+def _get_custom_pacing_children(subsection):
+    """
+    Return relative date items for the subsection and its children
+    """
+    items = [subsection]
+    section_date_items = []
+    while items:
+        next_item = items.pop()
+        num_weeks = next_item.fields['due_num_weeks'].read_from(next_item)
+        if num_weeks:
+            section_date_items.append((next_item.location, {'due': timedelta(weeks=num_weeks)}))
+        items.extend(next_item.get_children())
+    return section_date_items
+
+
 def extract_dates_from_course(course):
     """
     Extract all dates from the supplied course.
     """
     log.info('Extracting course dates for %s', course.id)
-    if course.self_paced:
+    if course.self_paced and custom_pls_is_active(course.id):
```

Comment:
This is a lot of duplicated code with what's below. Could you maybe combine these by moving the flag check lower?